packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -181,40 +147,33 @@ class NoopCacheClearStrategy implements CacheClearStrategy {
 }
 const noopCacheClearStrategy = new NoopCacheClearStrategy();
 
-export class StripeHelper {
+export class StripeHelper extends StripeHelperBase {
+  // Base class requirements
+  public override readonly stripe: Stripe;
+  protected override readonly stripeFirestore: StripeFirestore;
+  protected override readonly paymentConfigManager?: PaymentConfigManager;
+  protected override readonly redis?: ioredis.Redis;
+
   // Note that this isn't quite accurate, as the auth-server logger has some extras
   // attached to it in Hapi.
-  private log: Logger;
   private plansAndProductsCacheTtlSeconds: number;
   private stripeTaxRatesCacheTtlSeconds: number;
   private webhookSecret: string;
-  private redis: ioredis.Redis | undefined;
-  private statsd: StatsD;
   private taxIds: { [key: string]: string };
   private firestore: Firestore;
-  private stripeFirestore: StripeFirestore;
-  // TODO remove the ? when removing the SUBSCRIPTIONS_FIRESTORE_CONFIGS_ENABLED feature flag
-  private paymentConfigManager?: PaymentConfigManager;
   readonly googleMapsService: GoogleMapsService;
-  readonly stripe: Stripe;
   public currencyHelper: CurrencyHelper;
 
   /**
    * Create a Stripe Helper with built-in caching.
    */
   constructor(log: Logger, config: ConfigType, statsd: StatsD) {
-    this.log = log;
-    this.plansAndProductsCacheTtlSeconds = config.subhub.plansCacheTtlSeconds;
-    this.stripeTaxRatesCacheTtlSeconds =
-      config.subhub.stripeTaxRatesCacheTtlSeconds;
-    this.webhookSecret = config.subscriptions.stripeWebhookSecret;
-    this.taxIds = config.subscriptions.taxIds;
-    this.currencyHelper = Container.get(CurrencyHelper);
+    super(config, statsd, log);
```

Comment:
Dependency on config format again, ideally these should have their own config formats.

Reply:
I see now that this is like the above bit, and the Stripe object has its own type that config happens to meet.